Dockerfile

```diff
@@ -11,15 +11,14 @@ COPY vendor/ vendor/
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager github.com/open-policy-agent/gatekeeper/cmd/manager
 
 # Copy the controller-manager into a thin image
-FROM ubuntu:latest
+FROM ubuntu:18.04
```

Comment:
I think this should stay latest, so we don't need to keep on top of version changes to the ubuntu image

Reply:
I'd respectfully disagree. With `latest` label Ubuntu may ship an update that will be breaking things. 
There are a lot of blog posts about this. Here is a quote from [K8s docs 
](https://kubernetes.io/docs/concepts/configuration/overview/#container-images)
```
Note: You should avoid using the :latest tag when deploying containers in production as it is harder to track which version of the image is running and more difficult to roll back properly.
```

Reply:
The K8s doc you're citing is discussing which container to run on Kubernetes itself (i.e. the version tag to use in a Pod config). There I agree with the advice, as using the `:latest` tag means that you have given up control over what is running in production.

Using `:latest` for the base image means that new releases may have a different OS base. This doesn't break rollback/forward behavior because the Gatekeeper images themselves are still versioned. Further, new builds are validated via end-to-end tests using the built docker image.

Ubuntu's page for its [docker repo](https://hub.docker.com/_/ubuntu/?tab=description&page=1&name=18.) appears to suggest using latest is the recommended behavior:

`The ubuntu:latest tag points to the "latest LTS", since that's the version recommended for general use. The ubuntu:rolling tag points to the latest release (regardless of LTS status).`

Looking at the output from `docker image inspect`, one bug I'm seeing is that we are unable to see which specific base image is being used for a particular build. If possible, we should fix that to help with debugging, but we can also view the OS build by shelling into the container.

Using a 100% static root (e.g. sha256 hash or similar), is not favorable, as we would not pick up security patches.

Reply:
Sorry I am still not convinced on using `latest`. But since I am not a maintainer I will obey 😄 .
You are right on security patches thins,  any tag on docker image is mutable and can be re-pushed with new image that's how Ubuntu would ship security updates for previously released image tag e.g. `18.04`.
If you still want to use `latest` - I'll make the change.

Reply:
I think it's best if we stick with `latest`

Thank you for your flexibility, and for the layer/size reduction!